### PR TITLE
Adding UTF8 Marshalling API tests

### PR DESCRIPTION
**Pull request by tijoytom**

### CODE REVIEW

**Comment by tijoytom** [comment]
@stephentoub

File: src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/MarshalTests.cs [hunk 1]
```diff
@@ -483,5 +483,57 @@ public void IsComObject_NullObject_ThrowsArgumentNullException()
...
+
+            yield return new object[] { string.Empty, new byte[] { } };
+            
+            // bug fixed in coreclr
+            //yield return new object[] { null, new byte[] {}};
+        }
+
+        [Theory]
+        [MemberData(nameof(Encode_TestData))]
+        public static void TestUTF8Marshalling(string chars, byte[] expected)
```
Comment (tijoytom) [inline src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/MarshalTests.cs]: I dont need expected byte[] , will remove it.

File: src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/MarshalTests.cs [hunk 2]
```diff
@@ -483,5 +483,57 @@ public void IsComObject_NullObject_ThrowsArgumentNullException()
...
+
+            yield return new object[] { "\u00A4\u00D0aR|{AnGe\u00A3\u00A4", new byte[] { 0xC2, 0xA4, 0xC3, 0x90, 0x61, 0x52, 0x7C, 0x7B, 0x41, 0x6E, 0x47, 0x65, 0xC2, 0xA3, 0xC2, 0xA4 } };
+
+            yield return new object[] { "\uD800\uDC00\uD800\uDC00\uD800\uDC00", new byte[] { 0xF0, 0x90, 0x80, 0x80, 0xF0, 0x90, 0x80, 0x80, 0xF0, 0x90, 0x80, 0x80 } };
+
+            yield return new object[] { "\uD800\uDC00\u0065\uD800\uDC00", new byte[] { 0xF0, 0x90, 0x80, 0x80, 0x65, 0xF0, 0x90, 0x80, 0x80 } };
+
+            yield return new object[] { string.Empty, new byte[] { } };
+            
+            // bug fixed in coreclr
```
Comment (stephentoub) [inline src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/MarshalTests.cs]: Presumably we're waiting for an updated coreclr to be consumed into corefx?

**Comment by stephentoub** [comment]
@dotnet-bot test this plesae

**Comment by crummel** [comment]
@dotnet-bot test Linux x64 Release Build
@dotnet-bot test NETFX x86 Release Build
@dotnet-bot test OSX x64 Debug Build
@dotnet-bot test UWP CoreCLR x64 Debug Build

**Comment by crummel** [comment]
@tijoytom @stephentoub builds went through this time, for reference in case they age out again before you get a chance to look:
```
System\Runtime\InteropServices\MarshalTests.cs(528,35): error CS0117: 'Marshal' does not contain a definition for 'StringToCoTaskMemUTF8' [D:\j\workspace\windows-TGrou---2a8f9c29\src\System.Runtime.InteropServices\tests\System.Runtime.InteropServices.Tests.csproj]
System\Runtime\InteropServices\MarshalTests.cs(529,45): error CS0117: 'Marshal' does not contain a definition for 'PtrToStringUTF8' [D:\j\workspace\windows-TGrou---2a8f9c29\src\System.Runtime.InteropServices\tests\System.Runtime.InteropServices.Tests.csproj]
System\Runtime\InteropServices\MarshalTests.cs(535,29): error CS0117: 'Marshal' does not contain a definition for 'ZeroFreeCoTaskMemUTF8' [D:\j\workspace\windows-TGrou---2a8f9c29\src\System.Runtime.InteropServices\tests\System.Runtime.InteropServices.Tests.csproj]
```

**Comment by stephentoub** [comment]
Thanks, @crummel.

**Comment by tijoytom** [comment]
@dotnet-bot test OSX x64 Debug Build

**Comment by tijoytom** [comment]
@dotnet-bot test Linux x64 Release Build

File: src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/MarshalTests.cs (hunk 2 above)
Comment (tijoytom) [reply src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/MarshalTests.cs]: Yes

**Comment by karelz** [comment]
@tijoytom any update? There wasn't progress for 3 weeks on this PR ...

**Comment by tijoytom** [comment]
@dotnet-bot test Linux x64 Release Build

**Comment by tijoytom** [comment]
@stephentoub @ericstj 
Any idea why  PtrToStringUTF8 , ZeroFreeCoTaskMemUTF8 and StringToCoTaskMemUTF8 will be missing from Linux x64 and OSX x64 refrence assemblies  ? I don't see any ifdef or any special casing in ref or implementation . Is there some packaging magic that  need to be done?

**Comment by stephentoub** [comment]
The APIs only exist in netcoreapp.  The tests that use them need to be added to files that are only built in when targeting netcoreapp.

**Comment by tijoytom** [comment]
@dotnet-bot test Tizen armel Debug Build

**Comment by tijoytom** [comment]
@dotnet-bot test OSX x64 Debug Build

**Comment by tijoytom** [comment]
test Windows x64 Debug Build

**Comment by tijoytom** [comment]
test Windows x86 Release Build

**Comment by tijoytom** [comment]
test OSX x64 Debug Build

**Comment by tijoytom** [comment]
OSX x64 Debug Build builds are timing out ..

**Comment by tijoytom** [comment]
test Windows x64 Debug Build

**Comment by tijoytom** [comment]
test Windows x86 Release Build

**Comment by tijoytom** [comment]
@crummel  I am struggling to get his PR tru since the one of the other jobs fails with something like below ... I somehow got OSX tru , but Windows x64 Debug and x86 release is still failing despite many attempts
 If you can confirm it's some infrastructure issue i will merge the PR 
error : Unable to publish to 'https://helix.dot.net/api/2017-04-14/jobs' after 15 retries. Received status code: Forbidden Site Disabled

**Comment by tijoytom** [comment]
@dotnet-bot  test Windows x64 Debug Build

**Comment by tijoytom** [comment]
@dotnet-bot retest Windows x86 Release Build

**Comment by tijoytom** [comment]
@dotnet-bot retest Windows x86 Release Build

**Pull request merged**